cpp/src/join/hash_join.cu

```diff
@@ -235,7 +235,7 @@ hash_join::hash_join_impl::hash_join_impl(cudf::table_view const& build,
                                           null_equality compare_nulls,
                                           rmm::cuda_stream_view stream)
   : _is_empty{build.num_rows() == 0},
-    _hash_table{compute_hash_table_size(build.num_rows()),
+    _hash_table{0,
```

Comment:
Can't we just use direct init to avoid the issue? 
```suggestion
   _flattened_build_table{ flatten_nested_columns(...); }
   _hash_table{build_join_hash_table(_flattened_build_table, compare_nulls, stream)}
```